File: solutions/c/2.c
#include "euler.h"

int fib(int n) {
  if(n<3) return n;
  return fib(n-1) + fib(n-2);
}

int solution() {
  int sum = 0;
  int i, curr;
  for(i=1, curr=1; curr < 4000000; i++, curr=fib(i)) {
    if(curr % 2 == 0) {
      sum += curr;
    }
  }
  return sum;
}

File: solutions/ruby/2.rb
FibMemo = { 1 => 1, 2 => 2 }

def fib(n)
  FibMemo[n] ||= fib(n-1) + fib(n-2)
end

def solution
  sum, i, curr = 0, 1, 1
  while curr < 4_000_000
    curr = fib i
    sum += curr if curr.even?
    i   += 1
  end
  sum
end